scripts/zones/RoMaeve/npcs/Goblin_Footprint.lua

```diff
@@ -0,0 +1,66 @@
+-----------------------------------
+-- Area: Bostaunieux Oubliette
+--  NPC: Goblin Footprint
+-- Type: NPC
+-- !pos  111.806 -24.007 57.266 167
+-----------------------------------
+local ID = require("scripts/zones/RoMaeve/IDs")
+require("scripts/globals/pasteventwatcher")
+require("scripts/globals/quests")
+require("scripts/globals/missions")
+-----------------------------------
+---
+function onTrade(player,npc,trade)
+    local options = {}
+    if player:hasCompletedMission(WINDURST,dsp.mission.id.windurst.VAIN) == true then
+        options[1] = 1
+    end
+    if player:hasCompletedMission(WINDURST,dsp.mission.id.windurst.MOON_READING) == true then
+        options[2] = 1
+    end
+    if player:getQuestStatus(ZILART,dsp.quest.id.outlands.DIVINE_MIGHT) == QUEST_COMPLETED then
+        options[3] = 1
+    end
+    if player:hasCompletedMission(ASA,dsp.mission.id.asa.FOUNTAIN_OF_TROUBLE) == true then
+        options[4] = 1
+    end
+    local eventList = dsp.pasteventwatcher.goblinFootprint({options[1], options[2], options[3], options[4]})
+    if not next(options) == nil then
+        player:startEvent(6, eventList, 4294967294, 4294967294, 4294967294)
+    end
+end
+
+function onTrigger(player,npc)
+    if player:hasCompletedMission(WINDURST,dsp.mission.id.windurst.VAIN) == true or
+        player:hasCompletedMission(WINDURST,dsp.mission.id.windurst.MOON_READING) == true or
+        player:getQuestStatus(ZILART,dsp.quest.id.outlands.DIVINE_MIGHT) == QUEST_COMPLETED or
+        player:hasCompletedMission(ASA,dsp.mission.id.asa.FOUNTAIN_OF_TROUBLE) == true then
+        player:messageSpecial(ID.text.ITEMS_ITEMS_LA_LA)
+    else
+        player:messageSpecial(ID.text.NOTHING_OUT_OF_ORDINARY)
+    end
+end
+
+function onEventUpdate(player,csid,option)
+end
+
+function onEventFinish(player,csid,option)
+    local spokeWithGobbie = 1
+    if csid == 6 then
+        if option == 1 then
+            player:startEvent(2)            -- Vain
+        end
+        if option == 2 then
+            player:startEvent(4)            -- Moon Reading
+        end
+        if option == 3 then
+            player:startEvent(7,917,1408)   -- Divine Might
+        end
+        if option == 4 then
+            player:startEvent(71)           -- Fountain of Trouble
+        end
```

Comment:
```
local option_cses =
{
    [1] = function(player) player:startEvent(2) end,
    [2] = function(player) player:startEvent(4) end,
    [3] = function(player) player:startEvent(7,917,1408) end
}

onFinish
    if csid == 6
        if goblin_cses[option] then
            -- Bonus points if you use the option checks from before to double-check if the player meets the requirement before playing the event (ie: make sure client isn't attempting to forge packet with an option they're not qualified for)
            goblin_cses[option](player)
        end
    end
end
```

You can get a cleaner result if you're willing to play with `unpack` to handle event params, letting you kill the lambdas outright.

Reply:
this ^